File: Daily/sbgo/Leetcode119-杨辉三角.go
package sbgo

func getRow(rowIndex int) []int {
	pre := []int{}
	cur := []int{}
	for i := 0; i <= rowIndex; i++ {
		cur = make([]int, i+1)
		cur[0] = 1
		cur[i] = 1
		for j := 1; j < i; j++ {
			cur[j] = pre[j] + pre[j-1]
		}
		pre = cur
	}
	return cur
}

File: Leetcode/PY/Leetcode119-杨辉三角.py
from typing import List
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        pre = []
        cur = []
        for i in range(rowIndex + 1):
            cur = [0] * (i + 1)
            cur[0] = cur[i] = 1
            j = 1
            while j < i:
                cur[j] = pre[j] * 1 + pre[j-1] * 1
                j += 1
            pre = cur
        return cur